### PR TITLE
add file block store implementation

### PR DESCRIPTION
**Pull request by kevodwyer**

### CODE REVIEW

File: src/main/java/org/peergos/blockstore/FileBlockstore.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            }
+        }
+        LOG.info("Using FileBlockStore at location: " + root);
+    }
+
+    private Path getFilePath(Cid cid) {
+        String name = new StringBuilder(cid.toBase58()).reverse().toString();
+        int folderPrefixLength = 2;
+        Path path = Paths.get(name.substring(0, folderPrefixLength));
+        for (int i = 0; i < folderDepth; i++)
```
Comment (ianopolous) [inline src/main/java/org/peergos/blockstore/FileBlockstore.java:41]: Can we match what kubo does here?

File: src/main/java/org/peergos/blockstore/SQLiteBlockstore.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Boolean> has(Cid c) {
+        String selectSQL = "SELECT 1 FROM " + BLOCK_STORE_TABLE + " WHERE hash=?";
+        try (Connection connection = getConnection();
+             PreparedStatement pstmt = connection.prepareStatement(selectSQL)) {
+            pstmt.setString(1, c.toString());
```
Comment (ianopolous) [inline src/main/java/org/peergos/blockstore/SQLiteBlockstore.java:49]: We should enforce the encoding we use for cid here, or use the byte encoding

File: src/main/java/org/peergos/blockstore/FileBlockstore.java (hunk 1 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/blockstore/FileBlockstore.java]: done

File: src/main/java/org/peergos/blockstore/SQLiteBlockstore.java (hunk 2 above)
Comment (kevodwyer) [reply src/main/java/org/peergos/blockstore/SQLiteBlockstore.java]: SQLite block store removed for now due to size of sqlite jar file

**Review by ianopolous — APPROVED** [review]

**Pull request merged**